File: Baekjoon/Greedy/2847.py
# https://www.acmicpc.net/problem/2847

import sys
input = sys.stdin.readline

n = int(input())
score = []
for i in range(n):
    score.append(int(input()))
cnt = 0

for i in range(n - 1, 0, -1):
    if score[i] <= score[i - 1]:
        cnt += score[i - 1] - score[i] + 1
        score[i - 1] = score[i] - 1

print(cnt)

File: Baekjoon/수학/2847.cpp
// https://www.acmicpc.net/problem/2847

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> v(n);
    for (int i = 0; i < n; i++) {
        cin >> v[i];
    }
    int ans = 0;
    for (int i = n - 2; i >= 0; i--) {
        if (v[i] >= v[i + 1]) {
            ans += v[i] - v[i + 1] + 1;
            v[i] = v[i + 1] - 1;
        }
    }
    cout << ans << '\n';
    return 0;
}